metrics/middleware.go

```diff
@@ -0,0 +1,222 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var defaultMetricPath = "/metrics"
```

Comment:
Will this conflict with any existing prometheus metrics we have defined already (thinking of `goProbe` packages)?

Reply:
No. `/metrics` is used pretty much everywhere as the default path.